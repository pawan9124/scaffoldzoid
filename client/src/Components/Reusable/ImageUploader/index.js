import React, { useState } from "react";
import { IconButton } from "@material-ui/core";
import PhotoCamera from "@material-ui/icons/PhotoCamera";
import PropTypes from "prop-types";
import "./style.css";

const ImageUploader = (props) => {
  const [imageSatus, setImageStatus] = useState(false);
  //handle image changes
  const handleImageChange = (e) => {
    const dataFile = [];
    const rawFiles = [];
    const files = e.target.files;
    const length = files.length;
    rawFiles.push(files);
    return Promise.all(
      [].map.call(files, function (file) {
        return new Promise(function (resolve, reject) {
          var reader = new FileReader();
          reader.onloadend = function () {
            resolve({ result: reader.result, file: file });
          };
          reader.readAsDataURL(file);
        });
      })
    ).then((results) => {
      results.forEach((result) => {
        dataFile.push({ file: result.file, imagePreviewUrl: result.result });
        if (length === dataFile.length) {
          const newImages = [...dataFile];
          const newSentImages = [files];
          props.setSentImageFiles(newSentImages);
          props.setImageFiles(newImages);
          setImageStatus(true);
        }
      });
    });
  };
  return (
    <div className="image-upload">
      <label htmlFor="icon-button-file" title="Upload image">
        <IconButton color="primary" component="span">
          <PhotoCamera />
        </IconButton>
      </label>
      <input
        accept="image/*"
        id="icon-button-file"
        onChange={handleImageChange}
        type="file"
        style={{ display: "none" }}
      />
      {imageSatus && (
        <small className="register__validationErrors">
          Please click save button to save image
        </small>
      )}
    </div>
  );
};

ImageUploader.propTypes = {
  setSentImageFiles: PropTypes.func.isRequired,
  setImageFiles: PropTypes.func.isRequired,
};

export default ImageUploader;
