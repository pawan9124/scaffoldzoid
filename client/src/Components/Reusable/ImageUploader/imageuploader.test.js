import React from "react";
import { shallow } from "enzyme";
import { findByTestAttr, checkProps } from "../../../testUtils";
import ImageUploader from "./index";
import renderer from "react-test-renderer";

const props = {
  setSentImageFiles: jest.fn(),
  setImageFiles: jest.fn(),
};

describe("Taking snap shot", () => {
  it("Taking the snapshot of the ImageUploader Component", () => {
    const tree = renderer.create(<ImageUploader {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Checking the render of ImageUploader", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ImageUploader {...props} />);
  });
  it("Should Render a ImageUploader", () => {
    const button = findByTestAttr(wrapper, "imageUploaderComponent");
    expect(button.length).toBe(1);
  });
});

describe("Checking PropTypes", () => {
  it("Should Not throw a warning", () => {
    const expectedProps = props;

    const propsError = checkProps(ImageUploader, expectedProps);
    expect(propsError).toBeUndefined();
  });
});
