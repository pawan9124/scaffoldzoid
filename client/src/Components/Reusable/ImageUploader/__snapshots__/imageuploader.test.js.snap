// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taking snap shot Taking the snapshot of the ImageUploader Component 1`] = `
<div
  className="image-upload"
  data-test="imageUploaderComponent"
>
  <label
    htmlFor="icon-button-file"
    title="Upload image"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="3.2"
          />
          <path
            d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
          />
        </svg>
      </span>
    </span>
  </label>
  <input
    accept="image/*"
    id="icon-button-file"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;
